newspaper/extractors.py

```diff
@@ -175,48 +189,116 @@ def get_title(self, doc):
+            and len(title_text_h1) > len(title_text_fb):
+            title_text = title_text_h1
+            used_delimeter = True
+        elif filter_title_text_fb and filter_title_text_fb != filter_title_text \
+            and filter_title_text.startswith(filter_title_text_fb):
+            title_text = title_text_fb
+            used_delimeter = True
+        elif filter_title_text_h1 and filter_title_text_h1 != filter_title_text \
+            and filter_title_text.startswith(filter_title_text_h1) and words_title_text_h1 >= 4:
+            title_text = title_text_h1
+            used_delimeter = True
+
         # split title with |
-        if '|' in title_text:
-            title_text = self.split_title(title_text, PIPE_SPLITTER)
+        if not used_delimeter and '|' in title_text:
+            title_text = self.split_title(title_text, PIPE_SPLITTER, title_text_h1)
             used_delimeter = True
 
         # split title with -
         if not used_delimeter and '-' in title_text:
-            title_text = self.split_title(title_text, DASH_SPLITTER)
+            title_text = self.split_title(title_text, DASH_SPLITTER, title_text_h1)
             used_delimeter = True
 
         # split title with _
         if not used_delimeter and '_' in title_text:
-            title_text = self.split_title(title_text, UNDERSCORE_SPLITTER)
+            title_text = self.split_title(title_text, UNDERSCORE_SPLITTER, title_text_h1)
 
         # split title with /
         if not used_delimeter and '/' in title_text:
-            title_text = self.split_title(title_text, SLASH_SPLITTER)
+            title_text = self.split_title(title_text, SLASH_SPLITTER, title_text_h1)
             used_delimeter = True
 
         # split title with »
-        if not used_delimeter and '»' in title_text:
-            title_text = self.split_title(title_text, ARROWS_SPLITTER)
+        if not used_delimeter and u'»' in title_text:
+            title_text = self.split_title(title_text, ARROWS_SPLITTER, title_text_h1)
             used_delimeter = True
 
+        # deleted as too often removes relevant parts of the title
         # split title with :
-        if not used_delimeter and ':' in title_text:
-            title_text = self.split_title(title_text, COLON_SPLITTER)
-            used_delimeter = True
+        # if not used_delimeter and ':' in title_text:
+        #     title_text = self.split_title(title_text, COLON_SPLITTER, title_text_h1)
+        #     used_delimeter = True
 
         title = MOTLEY_REPLACEMENT.replaceAll(title_text)
+
+        # in some cases the final title is quite similar to title_text_h1
+        # (either it differs for case, for special chars, or it's truncated)
+        # in these cases, we prefer the title_text_h1
+        filter_title = filter_regex.sub('', title).lower()
+        if filter_title_text_h1 == filter_title or filter_title_text_h1.startswith(filter_title):
```

Comment:
If the `startswith` condition here is removed, it'll fix the error when unrelated stuff is appended to the titles coming from h1.

E.g. http://www.theglobeandmail.com/news/world/airasia-search/article22224253/
gets the title "Final manoeuvre of AirAsia flight raises questions over flying in storms Add to ..."